tests/python/unittest/test_tir_transform_convert_ssa.py

```diff
@@ -0,0 +1,255 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import tvm
+import tvm.testing
+from tvm.script import tir as T, ir as I
+
+
+class BaseBeforeAfter(tvm.testing.CompareBeforeAfter):
+    transform = tvm.tir.transform.ConvertSSA()
+
+
+class TestReuseInSequentialLetStmt(BaseBeforeAfter):
+    """De-dup sequential variable bindings"""
+
+    def before(self):
+        @T.prim_func
+        def func():
+            i1 = T.handle("int32")
+            with T.LetStmt(i1, var=i1):
+                T.evaluate(0)
+            with T.LetStmt(i1, var=i1):
```

Comment:
I am not sure if TVMScript support creation of non-SSA form, since  future parsers could choose to do renaming, or not support such case. 

It would be safer to do things with ir builder

Reply:
Good point, especially since all the before cases are malformed in some way.  The IRModule-wide checks should be safe, since the duplication is introduced after parsing, but both `TestReuseInSequentialLetStmt` and `TestReuseInNestedLetStmt` should be updated to prevent future improvements to the parser from fixing them before the test runs.

Reply:
Unit tests are now updated so that all SSA violations are introduced outside of TVMScript.